### ISSUE TITLE
CMake + distutils integration

### ISSUE THREAD

**Issue by oxullo** [libavg/libavg#751]
Attempt to conciliate cmake and distutils in order to standardize as much as possible python package's build and installation process.

Goals:
1. reduce as much as possible custom code
2. honor virtualenv (with a virtualenv path evaluation happening at install time)
3. add pip compatibility

**Comment by oxullo** [comment]
211e39818adda3df2685aab7871bde6099732022 is a first attempt to provide a proof of concept based on the idea described on this blog post: https://bloerg.net/2012/11/10/cmake-and-distutils.html
It solves points 1 and 2 but it leaves 3 out, since setup.py is called by make (via cmake).

Test with:
```
mktmpenv
cd /path/to/libavg
cmake .
make -j4
make install
DISPLAY=:0 python -c 'import libavg'
```

- Note: the branch break the tests
- Note: tested only under linux

**Comment by oxullo** [comment]
Referencing #753

**Comment by oxullo** [comment]
Fixed with #753
